src/ProducerEventProcessor/UploaderCaptureEventCollector.cpp

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "ProducerEventProcessor/UploaderCaptureEventCollector.h"
+
+#include "Introspection/Introspection.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/MakeUniqueForOverwrite.h"
+#include "OrbitBase/ThreadUtils.h"
+#include "capture.pb.h"
+#include "services.grpc.pb.h"
+
+using orbit_capture_uploader::DataReadiness;
+using orbit_grpc_protos::CaptureResponse;
+using orbit_grpc_protos::ClientCaptureEvent;
+
+namespace orbit_producer_event_processor {
+
+UploaderCaptureEventCollector::UploaderCaptureEventCollector() {
+  // Create an output stream which will be used to convert capture events into parts of a well
+  // formatted capture file. This should never fail.
+  capture_data_buffer_stream_ = std::make_unique<orbit_capture_file::BufferOutputStream>();
```

Comment:
Why a unique_ptr?

Reply:
Thanks for pointing out this. Changed to use just `BufferOutputStream`.